src/main/java/fi/okm/jod/yksilo/controller/profiili/ToimintoController.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024 The Finnish Ministry of Education and Culture, The Finnish
+ * The Ministry of Economic Affairs and Employment, The Finnish National Agency of
+ * Education (Opetushallitus) and The Finnish Development and Administration centre
+ * for ELY Centres and TE Offices (KEHA).
+ *
+ * Licensed under the EUPL-1.2-or-later.
+ */
+
+package fi.okm.jod.yksilo.controller.profiili;
+
+import fi.okm.jod.yksilo.domain.JodUser;
+import fi.okm.jod.yksilo.dto.IdDto;
+import fi.okm.jod.yksilo.dto.profiili.ToimintoDto;
+import fi.okm.jod.yksilo.dto.validationgroup.Add;
+import fi.okm.jod.yksilo.service.profiili.ToimintoService;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+@RestController
+@RequestMapping("/api/profiili/toiminnot")
+@RequiredArgsConstructor
+@Tag(name = "profiili")
+public class ToimintoController {
+  private final ToimintoService service;
+
+  @GetMapping
+  List<ToimintoDto> getAll(@AuthenticationPrincipal JodUser user) {
+    return service.findAll(user);
+  }
+
+  @PostMapping()
+  @ResponseStatus(HttpStatus.CREATED)
```

Comment:
These controller methods should have the openAPI annotations for the api documentation.